core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -103,21 +103,23 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
    */
   def resize(newSize: Int) {
     inLock(lock) {
-      val raf = new RandomAccessFile(file, "rw")
+      val pos = mmap.position()
```

Comment:
Looked a little bit into the test failure. The failure is caused by this. When a corrupted index file is found, that index file will be deleted from disk. And the recovery logic will call `index.resize()` which will create a new RandomAccessFile() in this case. But the patch removed this logic, so the deleted index file could not be found during the later on deletion.

The simplest fix would be have the following code piece before line 106.
```
      if (!file.exists()) {
        raf.close()
        raf = new RandomAccessFile(file, "rw")
      }
```

However, I do think the code logic should be more clear and the file creation should be more explicit. But we can do that in a separate patch given it is an existing issue, and this patch provides important performance improvement.